files/lib/form/FormBuilder.class.php

```diff
@@ -114,21 +114,14 @@ protected function buildAttributeList()
      * Initializes the values list based on the defined attribute list.
      */
     protected function initializeValues()
-    {
-        foreach ($this->buildAttributeList() as $name => $options) {
-            $this->valueList[$name] = null;
-        }
-    }
-
-    /**
-     * Is called when the form was submitted.
-     *
-     * @see \wcf\form\IForm::submit()
-     */
-    public function submit() 
-    {
-        parent::submit();
-    }
+	{
+		foreach ($this->buildAttributeList() as $name => $options) {
+		    //check if dont´t value/option exist
+			if(!isset($this->valueList[$name])){
+				$this->valueList[$name] = '';
```

Comment:
It won't matter whether the value is a string or `null` due to PHP's automatic type casting. I'll leave it at `null` for that to designate that there is in fact no value, not just an empty value.